### PR TITLE
chore(release): Allow pre-release for semantic branch prefixes

### PR DESCRIPTION
**Pull request by david-gomes5**
## What does this do?
Allow for pre-releases on semantic branch prefixes including:
- `feature/*`
- `fix/*`
- `chore/*`

Removed prefix:
- `crumbs-*`

All accepted prefixes can create a pre-release that'll use the current lib version and suffix it with `crumbs-<branch-name>.x` (`x` being the next available pre-release version)

## Relevant tickets / Documentation
- As part of [FE discussion on Slack](https://eatmarshmallows.slack.com/archives/C06636QJ6Q4/p1728383695116449?thread_ts=1728382767.690179&cid=C06636QJ6Q4)
## Testing
- Working on [CI](https://github.com/marshmallow-insurance/smores-react/actions/runs/11571164758/job/32208452065#step:5:107)

### CODE REVIEW

**Comment by david-gomes5** [comment]
Should update any docs that reference pre-releases.

File: .github/workflows/preview_bump_and_publish.yml [hunk 1]
```diff
@@ -10,14 +10,6 @@ jobs:
   preview-bump-and-publish:
     runs-on: ubuntu-latest
     steps:
-      - name: Check branch
-        run: |
-          if [[ "$GITHUB_REF" != "refs/heads/main" && "$GITHUB_REF" != refs/heads/crumbs-* ]]; then
-            echo "This job can only run on the main branch or branches starting with crumbs-"
-            exit 1
-          fi
-        shell: bash
```
Comment (rowanhen) [inline .github/workflows/preview_bump_and_publish.yml:19]: is it worth removing this? i think the early abort of the ci is quite nice & gives more info - can we tweak the if statement to allow for the newly allowed prefixes?

File: release.config.js [hunk 2]
```diff
@@ -1,12 +1,15 @@
...
+/**
+ * @type {import('semantic-release').GlobalConfig}
+ */
 const config = {
   branches: [
     'main',
     {
-      name: 'crumbs-*',
-      prerelease: true,
+      name: '(feature|fix|chore)/*',
```
Comment (rowanhen) [inline release.config.js:11]: we should also update the readme & frames with these same changes 🙏

File: .github/workflows/preview_bump_and_publish.yml (hunk 1 above)
Comment (david-gomes5) [reply .github/workflows/preview_bump_and_publish.yml:19]: Added the check in `release.config.js`

**Pull request merged**